### PR TITLE
Add support for return value indexing in `readEventArg` and `staticCall`

### LINKED ISSUE #396: Refine read event arg and static call api
Both read event arg and static call should be able to select the field of the result tuple by name or index.

Change readEventArgument parameter `argumentName` to be called `argument`, it should accept either string or number. Where the number is the index into the tuple.

We need to enhance staticCall so that there is an argument parameters after args. Argument is the number index for the result tuple.

Validation rules for out of range index and ensure that if the arg is a string that the name exists.

Take a look at the abi module, as some validation may already be in place.

### PR DESCRIPTION
**Pull request by zoeyTM**
fixes #396

### CODE REVIEW

File: packages/core/src/internal/reconciliation/futures/reconcileReadEventArgument.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ export function reconcileReadEventArgument(
   result = compare(
     future,
     "Argument name",
```
Comment (alcuadrado) [inline packages/core/src/internal/reconciliation/futures/reconcileReadEventArgument.ts:50]:
```suggestion
    "Argument name or index",
```

Comment (alcuadrado) [reply packages/core/src/internal/reconciliation/futures/reconcileReadEventArgument.ts:50]: I guess we should add an equivalent reconciliation rule to static call

**Comment by alcuadrado** [comment]
Good work! I left two minor comments about reconciliation

File: packages/core/src/internal/reconciliation/futures/reconcileReadEventArgument.ts (hunk 1 above)
Comment (zoeyTM) [reply packages/core/src/internal/reconciliation/futures/reconcileReadEventArgument.ts]: Done and done

**Pull request merged**